crypto/provider_core.c

```diff
@@ -1152,7 +1152,25 @@ static int provider_flush_store_cache(const OSSL_PROVIDER *prov)
     CRYPTO_THREAD_unlock(store->lock);
 
     if (!freeing)
-        return evp_method_store_flush(prov->libctx);
+        return evp_method_store_cache_flush(prov->libctx);
```

Comment:
One thing I found infuriating is that we do this for EVP stuff only, but not OSSL_STORE, OSSL_ENCODER or OSSL_DECODER.  I'm *not* addition the other implementation types in this PR, but to be complete, we really should, and also probably make a mechanism that helps us avoid this sort of hard coding, as it's just too easy to forget the gazillion places to touch as soon as a new (non_EVP) implementation type comes along